test/form/samples/default-identifier-deshadowing/dep.js

```diff
@@ -0,0 +1,5 @@
+export default function a() {
+	console.log('effect');
+  a = someGlobal;
+  return a();
+}
```

Comment:
I would remove the `console.log` statement as the function call should not be removed even without it. `a()` is basically equivalent to `someGlobal()` which can have side-effects.

Reply:
Sure, done.